wurzel/steps/step_docling/docling_step.py

```diff
@@ -49,8 +48,10 @@ class CleanMarkdownRenderer(HTMLRenderer):
     @staticmethod
     def render_html_block(token):
         # Remove comments like <!-- image -->
-        cleaned = re.sub(r"<!--.*?-->", "", token.content)
-        return cleaned.strip()
+        soup = BeautifulSoup(token.content, "html.parser")
+        for comment in soup.find_all(string=lambda text: isinstance(text, Comment)):
+            comment.extract()
+        return soup.decode_contents().strip()
 
```

Comment:
```suggestion
        soup = BeautifulSoup(token.content, "html.parser")
        comment.comment.extract()
        return soup.decode_contents().strip()

```
Should work the exact same;